unittest/src/fake/alarm/mod.rs

```diff
@@ -0,0 +1,78 @@
+//! Fake implementation of the Alarm API.
+//!
+//! Supports frequency and set_relative.
+//! Will call the upcall immediately.
```

Comment:
nitpick: use "schedule" rather than "call". This fake driver schedules the upcall immediately, and it is invoked on the next Yield call.